crates/sui-core/src/epoch/reconfiguration.rs

```diff
@@ -52,7 +52,7 @@ where
             unreachable!();
         }
 
-        self.state.halted.store(true, Ordering::SeqCst);
+        self.state.halt_validator();
         while !self.state.batch_notifier.ticket_drained() {
```

Comment:
Lets add a todo here to refactor to a condition variable pattern, this is a little fragile.